### DISCUSSION TITLE
CTkInput Dialogue box (functionalities)

### DISCUSSION THREAD

**Discussion by StandardV** [TomSchimansky/CustomTkinter#1216] [Ideas]
Hi everyone, I'm trying to make some new features for my program and suddenly I just noticed that there's something that I need from the CTkInput dialogue box but it's just not in there at all so I've look into the source for the ctk_input_dialog.py to see if it's  exist and I just don't know about it but I think I see something that can be improved in there:

- Ok and Cancel buttons seem to use the same function which will "get the text from the entry, release the grab on other windows and destroy the dialogue box", this is normal but if you think of it from the user point of view, they put some text into the entry, suddenly change their mind and don't want to do it anymore --> **press cancel to exit, destroy the widget and not storing the dialogue**. However for Ok button, they might want to enter information continuously, **press Ok and the dialogue box stay but store the entry into _user_input**

- The second thing is appear position of the dialogue box, it seem random in the top left for me majority of the time, it would be great if we are able to choose where it show up in our code as well, such as middle of main window, bottom corner..etc..I saw the implementation  of"button.place(relx=0.5, rely=0.5, anchor=tkinter.CENTER)" but it seem to just move the widget around so I've decided to not touch it

I could attempt to do it myself as well but as I don't have much experience with Github pull request or related functionalities so I've decided that I'll just post it here, thank you!

**Comment by ElectricCandlelight** [comment]
For the first point you're probably better implementing it yourself using CTkToplevel since input dialogue boxes aren't really meant to work like that.

To position a window on the screen at a specific location use .geometry("width x height + position_right + position_left")

**Reply by StandardV** [reply]
I understand, thank you!
